proj4/src/main/scala/geotrellis/proj4/mgrs/MGRS.scala

```diff
@@ -0,0 +1,485 @@
+package geotrellis.proj4.mgrs
+
+import scala.math.{Pi, cos, floor, pow, round, sin, sqrt, tan}
+
+/**
+ * A lat/long to Military Grid Reference System (MGRS) conversion utility
+ *
+ * This implementation is a port of proj4js/mgrs.  See https://github.com/proj4js/mgrs for more.
+ */
+object MGRS {
+  /**
+   * Converts a longitude/latitude position to an MGRS grid location.  The accuracy argument indicates
+   * the resolution of grid cells that should be used.  The default accuracy of 5 indicates 1m resolution,
+   * while an accuracy of 1 indicates 10km resolution, with power of 10 steps in between.
+   */
+  def longLatToMGRS(long: Double, lat: Double, accuracy: Int = 5): String = {
+    if (accuracy < 1 || accuracy > 5)
+      throw new IllegalArgumentException("Accuracy for MGRS conversion must be in the range 1 to 5")
+
+    encode(LLtoUTM(long, lat), accuracy)
+  }
+
+  /**
+   * Converts an MGRS grid specifier to a bounding box in long/lat coordinates.  The return value gives
+   * the bounding box as (left, bottom, right, top) positions.  If the accuracy is zero (i.e., no numerical
+   * location characters following the zone number, zone letter, and grid square identifiers), the bounding
+   * box will be degenerate; use mgrsToLongLat in that case.
+   */
+  def mgrsToBBox(mgrs: String): (Double, Double, Double, Double) = {
```

Comment:
This is obviously an `Extent` but we can't have `proj4` subproject depends on `vector` project. Which kind of means that this should be moved to `vector` even through this is dealing with a projecting. I favor the latter @lossyrob thoughts?